### PR TITLE
Implemented a converter from pull file to conf job configuration usin…

### PR DESCRIPTION
**Pull request by ibuenros**
…g templates.

### CODE REVIEW

**Comment by ibuenros** [comment]
@htran1 can you review? Thanks!

File: gobblin-runtime/build.gradle [hunk 1]
```diff
@@ -121,3 +121,39 @@ clean {
...
+  for (prop in props) {
+    if (!project.hasProperty(prop)) {
+      throw new RuntimeException("Missing property " + prop)
+    }
+  }
+
+  def templateFile = file(project.property(templatePath))
+  def templateClasspath = templateFile.getParent()
+
+  println templateClasspath
```
Comment (htran1) [inline gobblin-runtime/build.gradle]: Maybe print annotation like "template classpath:"

File: gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      System.out.println("Usage: PullFileToConfigConverter <pullFilesRootPath> <pullFilesToConvertGlob> resource:///<templatePath> <sysConfigPath> <outputDir>");
+      System.exit(1);
+    }
+
+    new PullFileToConfigConverter(new Path(args[0]), new Path(args[1]), new Path(args[2]), new File(args[3]), new Path(args[4]))
+        .convert();
+  }
+
+  public void convert() throws IOException {
+
```
Comment (htran1) [inline gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java]: Extra line.

File: gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        Sets.newHashSet(templateConfig.getStringList(DO_NOT_OVERRIDE_KEY)) : Sets.<String>newHashSet();
+
+    ConfigRenderOptions configRenderOptions = ConfigRenderOptions.defaults();
+    configRenderOptions = configRenderOptions.setComments(false);
+    configRenderOptions = configRenderOptions.setOriginComments(false);
+    configRenderOptions = configRenderOptions.setFormatted(true);
+    configRenderOptions = configRenderOptions.setJson(false);
+
+    for (FileStatus pullFile : pullFileFs.globStatus(this.fileGlobToConvert)) {
+
```
Comment (htran1) [inline gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java]: Extra line.

File: gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            }
+          } catch (ConfigException.NotResolved nre) {
+            // path is unresolved in config, will try again next iteration
+          }
+        }
+
+      } while (somethingChanged);
+
+      try {
+        Config currentOutputConfig = ConfigFactory.parseMap(outputConfigMap);
```
Comment (htran1) [inline gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java]: Can move currentOutputConfig out and reuse below in place of outputConfig.

File: gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      Path newPath = PathUtils.removeExtension(pullFile.getPath(),
+          PullFileLoader.DEFAULT_JAVA_PROPS_PULL_FILE_EXTENSIONS.toArray(new String[]{}));
+      newPath = PathUtils.addExtension(newPath, "conf");
+      newPath = new Path(this.outputPath, newPath.getName());
+
+      FSDataOutputStream os = outputFs.create(newPath);
+      os.write(rendered.getBytes(Charsets.UTF_8));
+      os.close();
+    }
+
```
Comment (htran1) [inline gobblin-runtime/src/main/java/gobblin/runtime/template/PullFileToConfigConverter.java]: Extra lines.

**Pull request merged**